src/main/java/com/textadventure/Lexer.java

```diff
@@ -26,8 +25,11 @@ private TokenType(String pattern) {
 			tokenPatternsBuffer.append(String.format("|(?<%s>%s)", tokenType.name(), tokenType.pattern));
 		Pattern tokenPatterns = Pattern.compile(new String(tokenPatternsBuffer.substring(1)));
```

Comment:
For tokenPatternsBuffer use `StringBuilder` instead of `StringBuffer`. StringBuffer's only advantage is that it's thread-safe but it comes with a performance cost. Here in this static function, it is much better to use StringBuilder which is really fast.
We'll have a separate discussion on thread-safety.